src/overlays/actors/ovl_En_Pr2/z_en_pr2.c

```diff
@@ -71,48 +92,677 @@ const ActorInit En_Pr2_InitVars = {
     (ActorFunc)EnPr2_Draw,
 };
 
-#endif
+typedef struct {
+    Actor actor;
+    char unk144[0xA];
+    s16 unk_14E;
+    char unk150[0xA];
+    s16 unk_15A;
+} TempStruct;
+
+void EnPr2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnPr2* this = THIS;
+
+    this->actor.targetMode = 3;
+    this->actor.hintId = 0x5B;
+    this->unk_1EC = 255;
+    this->actor.colChkInfo.health = 1;
+    this->actor.colChkInfo.damageTable = &sDamageTable;
+
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_pr_Skel_004188, &object_pr_Anim_004340, this->jointTable,
+                       this->morphtable, 5);
+    this->unk_1E0 = ENPR2_GET_F(&this->actor);
+    this->actor.colChkInfo.mass = 10;
+    Math_Vec3f_Copy(&this->unk_228, &this->actor.home.pos);
+
+    if (this->unk_1E0 == 2) {
+        this->unk_208 = ENPR2_GET_FF0(&this->actor) * 20.0f;
+    }
+    this->unk_1F4 = 255;
+    this->actor.shape.yOffset = 500.0f;
+    this->actor.shape.shadowScale = 12.0f;
+
+    if (this->unk_1E0 < 10) {
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 19.0f);
+        Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        this->unk_218 = -1;
+        this->unk_204 = 0.0f;
+
+        if (this->actor.parent != NULL) {
+            Actor* parent = this->actor.parent;
+
+            if (parent->update != NULL) {
+                if (parent->world.rot.y != 0) {
+                    this->unk_218 = parent->world.rot.y - 1;
+                }
+            }
+        } else if (this->actor.world.rot.z != 0) {
+            this->unk_218 = this->actor.world.rot.z - 1;
+            this->actor.world.rot.z = 0;
+        }
+
+        if (this->unk_1E0 == 3) {
+            if (this->actor.parent != NULL) {
+                Actor* parent = this->actor.parent;
+
+                if (parent->update != NULL) {
+                    this->unk_1C8 = ((TempStruct*)parent)->unk_15A;
```

Comment:
Did some digging in KZ, and this parent actor looks to be of type `En_Encount1` which makes sense as it can act as a skullfish spawner.

Reply:
This doesn't seem to be correct, or not in all cases anyway. encount1 has SkelAnime at 0x144, which means this is skelanime->0x16, which is `/* 0x14 */ f32 animLength;`. unk_14E also doesn't make sense.

Reply:
Are you sure you checked at the right actor?
I just checked EnEncount1 in m2c, and it doesn't have a SkelAnime member (it wouldn't even fit in the struct).
unk_15A seems to be related to its params and unk_14E seems to be real too.

Reply:
Don't forget to use `En_Encount1` here instead of `TempStruct`